lib/logstash/inputs/dynamodb.rb

```diff
@@ -0,0 +1,359 @@
+  config :perform_stream, :validate => :boolean, :default => true
+
+  # Number of read operations per second to perform when scanning the specified table.
+  config :read_ops, :validate => :number, :default => 1
+
+  # Number of threads to use when scanning the specified table
+  config :number_of_scan_threads, :validate => :number, :default => 1
+
+
+  # Number of threads to write to the logstash queue when scanning the table
+  config :number_of_write_threads, :validate => :number, :default => 1
+
+  # Configuation for how the logs will be transferred.
+  # plain is simply pass the message along without editing it.
+  # dynamodb will return just the data specified in the view_format in dynamodb format.
+    # For more information see: docs.aws.amazon.com/amazondynamodb/latest/developerguide/DataFormat.html
+  # json_drop_binary will return just the data specified in the view_format in JSON while not including any binary values that were present.
+  # json_binary_as_text will return just the data specified in the view_format in JSON while including binary values as base64-encoded text.
+  config :log_format, :validate => [LF_PLAIN, LF_DYNAMODB, LF_JSON_NO_BIN, LF_JSON_BIN_AS_TEXT], :default => "plain"
+
+  public
+  def build_credentials
+    if not @aws_access_key_id.to_s.empty? and not @aws_secret_access_key.to_s.empty?
+      @logger.info("Using static credentials: " + @aws_access_key_id + ", " + @aws_secret_access_key)
+      basic = AmazonCredentials::BasicAWSCredentials.new(@aws_access_key_id, @aws_secret_access_key)
+      return AmazonCredentials::StaticCredentialsProvider.new(basic)
+    else
+      @logger.info("Using default provider chain")
+      return AmazonCredentials::DefaultAWSCredentialsProviderChain.new()
+    end # if neither aws access keys
+  end # def build_credentials
+
+  public
+  def register
+    if not LogStash::Environment.jruby?
+      raise(LogStash::PluginLoadingError, "KCL requires JRuby for streams processing")
+    end # if not LogStash::Environment.jruby?
+
+
+    LogStash::Logger.setup_log4j(@logger)
+
+    @host = Socket.gethostname
+    @logger.info("Tablename: " + @table_name)
+    @queue = SizedQueue.new(20)
+    @credentials = build_credentials()
+    @logger.info("Checkpointer: " + @checkpointer)
+
+    if @perform_scan and @view_type === VT_OLD_IMAGE
+      raise(LogStash::ConfigurationError, "Cannot perform scan with view type: " + @view_type + " configuration")
+    end
+    if @view_type === VT_ALL_IMAGES and (not @log_format === LF_PLAIN)
+      raise(LogStash::ConfigurationError, "Cannot show view_type: " + @view_type + ", with log_format: " + @log_format)
+    end
+
+    #Create DynamoDB Client
+    @dynamodb_client = AmazonDynamoDB::AmazonDynamoDBClient.new(@credentials)
+
+    @logger.info(@dynamodb_client.to_s)
+
+    @dynamodb_client.setEndpoint(@endpoint)
+    @logger.info("DynamoDB endpoint: " + @endpoint)
+
+    @key_schema = Array.new
```

Comment:
This, until line ```192``` is a java code within the ruby world, I was wondering if makes sense to wrap it inside some Ruby classes to be more Ruby friendly, or if there are some dynamodb clients out there that you can leverage not to actually have to go direct to the java world, what do you think?